kernel/thread.c

```diff
@@ -141,6 +142,7 @@ bool k_is_in_isr(void)
 {
 	return arch_is_in_isr();
 }
+EXPORT_SYMBOL(k_is_in_isr);
```

Comment:
Again, this is a global symbol - there is no implicit dependency on llext for `k_is_in_isr()`.

I can understand the case where a particular private or static function needs to be exported for llext (for visibility), but for globals, standard C functions, can we just put them into `llext_export.c` as well?

Reply:
well, I think I'd disagree here. Think about adding a new function that you want to export. You add it where it belongs and you just add that one single line to that file to indicate that it's exported. You don't want to go and edit a common global file for all these exports. Also this is the same as what the Linux kernel does. We agreed with libc symbols because those functions can have multiple implementations, but "normal" functions, that only have one implementation each, should export the symbol themselves.

Reply:
Linux is built around the premise of *most* optional components being tristate selectable. Zephyr is not. There is a huge difference.

What you're describing is a treewide change introducing a dependency on a header that is experimental. There is a process for that. Personally, I would prefer to see a less invasive approach for this feature until it is stable.

It would be different if it were just part of the regular Zephyr API and available via `<zephyr/kernel.h>` or something along those lines, but realistically, it is still experimental.